control-plane/rest/service/src/v0/block_devices.rs

```diff
@@ -0,0 +1,36 @@
+use super::*;
+use mbus_api::v0::ListBlockDevices;
+
+pub(super) fn configure(cfg: &mut paperclip::actix::web::ServiceConfig) {
+    cfg.service(list_block_devices);
+}
+
+// List block devices takes a query parameter 'all' which is used to determine
+// whether to return all found devices or only those that are usable.
+// Omitting the query parameter will result in all block devices being shown.
+//
+// # Examples
+// List only usable block devices with query parameter:
+//      curl -X GET "https://localhost:8080/v0/nodes/mayastor/block_devices/list?all=false" \
+//      -H  "accept: application/json"
+//
+// List all block devices with query parameter:
+//      curl -X GET "https://localhost:8080/v0/nodes/mayastor/block_devices/list?all=true" \
+//      -H  "accept: application/json" -k
+//
+// List all block devices without query parameter:
+//      curl -X GET "https://localhost:8080/v0/nodes/mayastor/block_devices/list" \
+//      -H  "accept: application/json" -k
+//
+#[get("/v0", "/nodes/{node}/block_devices/list", tags(BlockDevices))]
+async fn list_block_devices(
+    web::Query(info): web::Query<ListBlockQueryParams>,
+    web::Path(node): web::Path<NodeId>,
+) -> Result<Json<Vec<BlockDevice>>, RestError> {
+    RestRespond::result(Ok(MessageBus::list_block_devices(ListBlockDevices {
+        node,
+        all: info.all.unwrap_or(true),
```

Comment:
how does this look like in the spec, does it show that the default is true?

Reply:
Unfortunately it doesn't.

Reply:
I think here that we want the default to be false. That was the original intention, and is certainly the case for mayastor-client

Reply:
The reason I chose to default to `true` is because we don't know who may be querying for this. It could potentially be an operator written by somebody else who then implements their own filtering criteria.